python/plugins/processing/core/parameters.py

```diff
@@ -434,14 +441,19 @@ def getAsScriptCode(self):
 class ParameterNumber(Parameter):
 
     def __init__(self, name='', description='', minValue=None, maxValue=None,
-                 default=0.0, optional=False):
-        Parameter.__init__(self, name, description, optional)
-        try:
-            self.default = int(unicode(default))
-            self.isInteger = True
-        except:
-            self.default = float(default)
+                 default=None, optional=False):
```

Comment:
I'd say, go for None everywhere. If an external provider expects an optional boolean parameter and provides good defaults for it, why overwrite it with True?